backend/substrapp/tasks/k8s_backend.py

```diff
@@ -100,6 +101,14 @@ def watch_pod(name, watch_init_container=False):
                 pretty=True
             )
 
+            # Handle pod error not linked with containers
+            if api_response.status.phase == 'Failed' or (api_response.status.reason and
+               'Evicted' in api_response.status.reason):
+                error_pod = f'Pod failed reason {api_response.status.reason})'
+                attempt += 1
```

Comment:
IMO we should immediately exit the function here because the POD will not recover even after multiple attempts: the time.sleep is 0.2 secs and if I interpret the [doc](https://kubernetes.io/docs/concepts/workloads/pods/pod-lifecycle/#restart-policy) correctly the first restart of the pod will happen after 10 sec.

Reply:
Yes, thank you for digging in this documentation !